rest-api-spec/schema.json

```diff
@@ -0,0 +1,242 @@
+{
```

Comment:
Can we include this as part of the work to make the validation happen in gradle?

Feel its better to discuss this on that PR.

Reply:
Was thinking it may be useful for folks to have beforehand, without needing to wait on a gradle task? [IntelliJ supports configuring a JSON schema](https://www.jetbrains.com/help/idea/json.html#) that might be useful in the meantime

Reply:
Doesn't hurt to throw this in there. We'll likely move it when we integrate the validation bit though.

Reply:
I started an [implementation](https://github.com/jakelandis/elasticsearch/tree/rest-spec-validation-gradle
) of a gradle task to validate the spec based on https://github.com/networknt/json-schema-validator.

I ran it against the core and x-pack specs and found 3 validation failures. 

Validating JSON spec [cat.thread_pool.json]
$.cat.thread_pool.params.size.options[0]: does not match the regex pattern ^[a-zA-Z_]+$

Validating JSON spec [ml.validate_detector.json]
$.ml.validate_detector.documentation.url: null found, string expected

Validating JSON spec [ml.validate.json]
$.ml.validate.documentation.url: null found, string expected

The rest pass!  

I should be able to clean up the task and get it wired in pretty soon after this gets merged.

Reply:
@jakelandis nice! 

Those failures are expected; I opened https://github.com/elastic/elasticsearch/issues/54253 for the ML failures. I’ll look into the cat thread pool failure and open an issue for it - I want to check to see what the intended behaviour of passing an empty size is